src/_pytest/fixtures.py

```diff
@@ -1448,6 +1449,20 @@ def deduplicate_names(*seqs: Iterable[str]) -> tuple[str, ...]:
     return tuple(dict.fromkeys(name for seq in seqs for name in seq))
 
 
+def default_arg_warn(nodeid: str, function_name, param_name, param_val) -> None:
+    msg = (
+        f"Test function '{function_name}' has a default argument '{param_name}={param_val}'.\n"
+    )
+    warnings.simplefilter("always", PytestDefaultArgumentWarning)
```

Comment:
I still think this isn't a good idea, what's the rationale for it? With it, a

```ini
pytestwarnings =
    ...
    ignore::_pytest.warning_types.PytestDefaultArgumentWarning
```

in pytest's configuration would have no effect at all...

(Also, the type should probably be exposed as `pytest.PytestDefaultArgumentWarning` just like `pytest.PytestUnknownMarkWarning` is?